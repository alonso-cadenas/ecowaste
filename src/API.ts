/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type CreateItemInput = {
  id?: string | null,
  name: string,
  description?: string | null,
  imageUrl?: string | null,
  category: Category,
  alternatives?: Array< string > | null,
};

export enum Category {
  COMPOSTABLE = "COMPOSTABLE",
  RECYCLABLE = "RECYCLABLE",
  REUSABLE = "REUSABLE",
  HAZARDOUS = "HAZARDOUS",
  TRIMMINGS = "TRIMMINGS",
  LANDFILL = "LANDFILL",
  DONATIONS = "DONATIONS",
}


export type ModelItemConditionInput = {
  name?: ModelStringInput | null,
  description?: ModelStringInput | null,
  imageUrl?: ModelStringInput | null,
  category?: ModelCategoryInput | null,
  alternatives?: ModelStringInput | null,
  and?: Array< ModelItemConditionInput | null > | null,
  or?: Array< ModelItemConditionInput | null > | null,
  not?: ModelItemConditionInput | null,
};

export type ModelStringInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export enum ModelAttributeTypes {
  binary = "binary",
  binarySet = "binarySet",
  bool = "bool",
  list = "list",
  map = "map",
  number = "number",
  numberSet = "numberSet",
  string = "string",
  stringSet = "stringSet",
  _null = "_null",
}


export type ModelSizeInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
};

export type ModelCategoryInput = {
  eq?: Category | null,
  ne?: Category | null,
};

export type Item = {
  __typename: "Item",
  id: string,
  name: string,
  description?: string | null,
  imageUrl?: string | null,
  category: Category,
  alternatives?: Array< string > | null,
  createdAt: string,
  updatedAt: string,
  owner?: string | null,
};

export type UpdateItemInput = {
  id: string,
  name?: string | null,
  description?: string | null,
  imageUrl?: string | null,
  category?: Category | null,
  alternatives?: Array< string > | null,
};

export type DeleteItemInput = {
  id: string,
};

export type CreateLocationInput = {
  id?: string | null,
  name: string,
  description?: string | null,
};

export type ModelLocationConditionInput = {
  name?: ModelStringInput | null,
  description?: ModelStringInput | null,
  and?: Array< ModelLocationConditionInput | null > | null,
  or?: Array< ModelLocationConditionInput | null > | null,
  not?: ModelLocationConditionInput | null,
};

export type Location = {
  __typename: "Location",
  id: string,
  name: string,
  description?: string | null,
  createdAt: string,
  updatedAt: string,
  owner?: string | null,
};

export type UpdateLocationInput = {
  id: string,
  name?: string | null,
  description?: string | null,
};

export type DeleteLocationInput = {
  id: string,
};

export type ModelItemFilterInput = {
  id?: ModelIDInput | null,
  name?: ModelStringInput | null,
  description?: ModelStringInput | null,
  imageUrl?: ModelStringInput | null,
  category?: ModelCategoryInput | null,
  alternatives?: ModelStringInput | null,
  and?: Array< ModelItemFilterInput | null > | null,
  or?: Array< ModelItemFilterInput | null > | null,
  not?: ModelItemFilterInput | null,
};

export type ModelIDInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export type ModelItemConnection = {
  __typename: "ModelItemConnection",
  items?:  Array<Item | null > | null,
  nextToken?: string | null,
};

export type SearchableItemFilterInput = {
  id?: SearchableIDFilterInput | null,
  name?: SearchableStringFilterInput | null,
  description?: SearchableStringFilterInput | null,
  imageUrl?: SearchableStringFilterInput | null,
  alternatives?: SearchableStringFilterInput | null,
  and?: Array< SearchableItemFilterInput | null > | null,
  or?: Array< SearchableItemFilterInput | null > | null,
  not?: SearchableItemFilterInput | null,
};

export type SearchableIDFilterInput = {
  ne?: string | null,
  gt?: string | null,
  lt?: string | null,
  gte?: string | null,
  lte?: string | null,
  eq?: string | null,
  match?: string | null,
  matchPhrase?: string | null,
  matchPhrasePrefix?: string | null,
  multiMatch?: string | null,
  exists?: boolean | null,
  wildcard?: string | null,
  regexp?: string | null,
  range?: Array< string | null > | null,
};

export type SearchableStringFilterInput = {
  ne?: string | null,
  gt?: string | null,
  lt?: string | null,
  gte?: string | null,
  lte?: string | null,
  eq?: string | null,
  match?: string | null,
  matchPhrase?: string | null,
  matchPhrasePrefix?: string | null,
  multiMatch?: string | null,
  exists?: boolean | null,
  wildcard?: string | null,
  regexp?: string | null,
  range?: Array< string | null > | null,
};

export type SearchableItemSortInput = {
  field?: SearchableItemSortableFields | null,
  direction?: SearchableSortDirection | null,
};

export enum SearchableItemSortableFields {
  id = "id",
  name = "name",
  description = "description",
  imageUrl = "imageUrl",
  alternatives = "alternatives",
}


export enum SearchableSortDirection {
  asc = "asc",
  desc = "desc",
}


export type SearchableItemConnection = {
  __typename: "SearchableItemConnection",
  items?:  Array<Item | null > | null,
  nextToken?: string | null,
  total?: number | null,
};

export type ModelLocationFilterInput = {
  id?: ModelIDInput | null,
  name?: ModelStringInput | null,
  description?: ModelStringInput | null,
  and?: Array< ModelLocationFilterInput | null > | null,
  or?: Array< ModelLocationFilterInput | null > | null,
  not?: ModelLocationFilterInput | null,
};

export type ModelLocationConnection = {
  __typename: "ModelLocationConnection",
  items?:  Array<Location | null > | null,
  nextToken?: string | null,
};

export type CreateItemMutationVariables = {
  input: CreateItemInput,
  condition?: ModelItemConditionInput | null,
};

export type CreateItemMutation = {
  createItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type UpdateItemMutationVariables = {
  input: UpdateItemInput,
  condition?: ModelItemConditionInput | null,
};

export type UpdateItemMutation = {
  updateItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type DeleteItemMutationVariables = {
  input: DeleteItemInput,
  condition?: ModelItemConditionInput | null,
};

export type DeleteItemMutation = {
  deleteItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type CreateLocationMutationVariables = {
  input: CreateLocationInput,
  condition?: ModelLocationConditionInput | null,
};

export type CreateLocationMutation = {
  createLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type UpdateLocationMutationVariables = {
  input: UpdateLocationInput,
  condition?: ModelLocationConditionInput | null,
};

export type UpdateLocationMutation = {
  updateLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type DeleteLocationMutationVariables = {
  input: DeleteLocationInput,
  condition?: ModelLocationConditionInput | null,
};

export type DeleteLocationMutation = {
  deleteLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type GetItemQueryVariables = {
  id: string,
};

export type GetItemQuery = {
  getItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type ListItemsQueryVariables = {
  filter?: ModelItemFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListItemsQuery = {
  listItems?:  {
    __typename: "ModelItemConnection",
    items?:  Array< {
      __typename: "Item",
      id: string,
      name: string,
      description?: string | null,
      imageUrl?: string | null,
      category: Category,
      alternatives?: Array< string > | null,
      createdAt: string,
      updatedAt: string,
      owner?: string | null,
    } | null > | null,
    nextToken?: string | null,
  } | null,
};

export type SearchItemsQueryVariables = {
  filter?: SearchableItemFilterInput | null,
  sort?: SearchableItemSortInput | null,
  limit?: number | null,
  nextToken?: string | null,
  from?: number | null,
};

export type SearchItemsQuery = {
  searchItems?:  {
    __typename: "SearchableItemConnection",
    items?:  Array< {
      __typename: "Item",
      id: string,
      name: string,
      description?: string | null,
      imageUrl?: string | null,
      category: Category,
      alternatives?: Array< string > | null,
      createdAt: string,
      updatedAt: string,
      owner?: string | null,
    } | null > | null,
    nextToken?: string | null,
    total?: number | null,
  } | null,
};

export type GetLocationQueryVariables = {
  id: string,
};

export type GetLocationQuery = {
  getLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type ListLocationsQueryVariables = {
  filter?: ModelLocationFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListLocationsQuery = {
  listLocations?:  {
    __typename: "ModelLocationConnection",
    items?:  Array< {
      __typename: "Location",
      id: string,
      name: string,
      description?: string | null,
      createdAt: string,
      updatedAt: string,
      owner?: string | null,
    } | null > | null,
    nextToken?: string | null,
  } | null,
};

export type OnCreateItemSubscription = {
  onCreateItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnUpdateItemSubscription = {
  onUpdateItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnDeleteItemSubscription = {
  onDeleteItem?:  {
    __typename: "Item",
    id: string,
    name: string,
    description?: string | null,
    imageUrl?: string | null,
    category: Category,
    alternatives?: Array< string > | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnCreateLocationSubscription = {
  onCreateLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnUpdateLocationSubscription = {
  onUpdateLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};

export type OnDeleteLocationSubscription = {
  onDeleteLocation?:  {
    __typename: "Location",
    id: string,
    name: string,
    description?: string | null,
    createdAt: string,
    updatedAt: string,
    owner?: string | null,
  } | null,
};
