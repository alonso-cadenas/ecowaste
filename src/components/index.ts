export * from './alternatives/alternatives';
export * from './appointment/appointment';
export * from './header/header';
export * from './item-list/item-list';
export * from './location/location';
export * from './search/search';
