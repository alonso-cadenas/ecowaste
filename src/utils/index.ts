export * from './getMessage/getMessage';
export * from './getSearchResults/getSearchResults';
export * from './goToSearchPage/goToSearchPage';
export * from './sanitizeInput/sanitizeInput';
