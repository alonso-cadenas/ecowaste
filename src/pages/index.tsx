import styles from '../styles/Home.module.css';
import { Header, Search } from '../components';
import Head from 'next/head';

export default function HomePage() {
  return (
    <section className={styles.container}>
      <Head>
        <title>Home - EcoWaste</title>
        <meta
          name="description"
          content="Learn about sustainable waste disposal."
        />
        <meta property="og:title" content="Home - EcoWaste" />
        <meta
          property="og:description"
          content="Learn about sustainable waste disposal."
        />
        <meta property="og:url" content="https://ecowaste.com" />
      </Head>

      <Header />

      <main className={styles.main}>
        <h1 className={styles.title}>EcoWaste</h1>
        <h2>Learn how to dispose of your household items</h2>
        <Search />
        <img id={'banner'} alt={'Banner'} src={'/banner.jpeg'} />
      </main>
    </section>
  );
}
