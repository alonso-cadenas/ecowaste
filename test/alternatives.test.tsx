import { Alternatives } from '../src/components';
import { render } from './testUtils';

describe('Alternatives', () => {
  test('renders without alternatives', () => {
    const { asFragment } = render(<Alternatives alternatives={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with alternatives', () => {
    const { asFragment } = render(
      <Alternatives
        alternatives={['metal spoon', 'plastic spoon', 'wooden spoon']}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
