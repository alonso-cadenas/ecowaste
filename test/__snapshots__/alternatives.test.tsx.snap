// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alternatives renders with alternatives 1`] = `
<DocumentFragment>
  <section
    id="alternatives"
  >
    <h2>
      Alternatives
    </h2>
    <ul>
      <li>
        metal spoon
      </li>
      <li>
        plastic spoon
      </li>
      <li>
        wooden spoon
      </li>
    </ul>
  </section>
</DocumentFragment>
`;

exports[`Alternatives renders without alternatives 1`] = `
<DocumentFragment>
  <section
    id="alternatives"
  >
    <h2>
      Alternatives
    </h2>
    <ul />
  </section>
</DocumentFragment>
`;
